lib/service/navigator_middleware.dart

```diff
@@ -13,67 +12,64 @@ class NavigatorMiddleware<R extends Route<dynamic>> extends RouteObserver<R> {
     this.onReplace,
     this.onRemove,
   }) : _stack = [];
-
+  final GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
```

Comment:
Why do we need this navigator key?

Reply:
It's Optional for setting up with Navigator 2.0 routing. As when using middleware, if one would want to navigate, they can use navigator key. It's also useful for Nested Navigation. 
So, I should have quoted this in Comments.

Reply:
@payam-zahedi I have updated the code to the best of my knowledge. please care to look into it and update me for any other required changes. Thanks.